service/service.go

```diff
@@ -306,8 +335,7 @@ func New(config Config) (*Service, error) {
 	s := &Service{
 		Version:                 versionService,
 		bootOnce:                sync.Once{},
-		clusterController:       clusterController,
-		machinePoolController:   machinePoolController,
+		controllers:             controllers,
```

Comment:
Having an slice of controllers allow us to add controllers _on demand_ depending on the enabled feature flags.

Reply:
Sounds useful!